Module/client/src/components/app.jsx

```diff
@@ -18,17 +21,22 @@ class App extends React.Component {
     axios.get('/listinginfo').then((response) => {
       let { id } = this.state;
       if (!id || (id > 100) || (id < 0)) { id = 0; }
-      this.setState({ data: response.data[id] });
+      this.setState({ data: response.data[id], loading: false });
```

Comment:
What does this do?

Reply:
It ensures the page will not render until the get request is completed and there will only be one get request... if theres a better way of doing this lmk. I originally assumed this was the job of componentDidMount()

Reply:
Unsure which is better honestly, might just be preference. I usually go with componentDidMount(). Up to you.